ksso/credstore.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import logging
+import pickle
+from datetime import datetime
+from typing import Optional
+
+import keyring
+
+from ksso.aws import AWSCredentials
+
+logger = logging.getLogger(__name__)
+
+KEYRING_SERVICE = "ksso-aws-credentials"
+
+
+def get_cache_key(client_id: str, role_arn: str) -> str:
+    """Generate a unique key for storing credentials in the keyring."""
+    return f"{client_id}||{role_arn}"
+
+
+def load_credentials(client_id: str, role_arn: str) -> Optional[AWSCredentials]:
```

Comment:
Since python 3.10 you no longer need to use the Optional keyword:


```suggestion
def load_credentials(client_id: str, role_arn: str) -> AWSCredentials | None:
```

Reply:
Done